docs: fix links to docs and pricing thoughout the code

thomas-dkmt opened the pull request:
Across various pages of the Xen Orchestra documentation, links to other pages within the documentation were using outdated syntax. External links have been replaced with internal ones where applicable, and all other links have been updated to the current syntax.

Additionally, links to the XOA pricing page now correctly point to the [latest version](https://vates.tech/pricing-and-support/) instead of the old one.

thomas-dkmt committed:
doc(XO): Correct external links to the XO doc and pricing page

On various pages of the documentation, links to other pages followed the old syntax. External links have been replaced with internal links where appropriate, and the other links were updated with the current syntax.

In addition, links to the pricing page now point to the current version instead of the old pricing page

Signed-off-by: Thomas Moraine <thomas.moraine@vates.tech>

nathanael-h commented on docs/docs/assets/stats.png:
Is this image related to this PR that focuses on URLs?

thomas-dkmt replied:
The image is referenced [here](https://github.com/vatesfr/xen-orchestra/pull/8558/files#diff-24556907336a943ff2942a5765b889a8ae41be882ddb1a15101b7d5b11b7d79f).

It was originally an external link but I changed it into an internal one.

nathanael-h replied:
Oh perfect!

nathanael-h commented on docs/docs/general-troubleshooting.md:
This pages has no sidebar, and the link L21 is not correct, see screenshot 
![image](https://github.com/user-attachments/assets/6f34094d-e91b-4810-9e08-c5a74540df0a)

nathanael-h commented on docs/docs/airgap.md:
```diff
@@ -15,11 +15,11 @@ In this scenario, you will need a QA/pre-production XCP-ng pool with Internet ac
 ### Deployment
 
-[Follow the standard procedure](https://xen-orchestra.com/docs/installation.html) to deploy the XOA on your pool with Internet access.
+[Follow the standard procedure](installation) to deploy the XOA on your pool with Internet access.
```
There might something to fix in config of docurausus of elsewhere, short links are not as expected : 
Should be `https://docs.xen-orchestra.com/installation`
Is `https://xen-orchestra.com/docs/installation.html`

![image](https://github.com/user-attachments/assets/c3609e23-2807-43be-b619-a45642b2af62)

nathanael-h replied:
Forget this comment, it should be ok.

nathanael-h reviewed (commented):
Good PR! I like when old things are cleaned :+1: :+1: :+1: A fix might be needed for links, and small other things. Call me if needed.

lsouai-vates commented on docs/docs/airgap.md:
```diff
@@ -15,11 +15,11 @@ In this scenario, you will need a QA/pre-production XCP-ng pool with Internet ac
+[Follow the standard procedure](installation) to deploy the XOA on your pool with Internet access.
 
-Make sure that your appliance is [properly registered](https://xen-orchestra.com/docs/installation.html#registration) and [up-to-date](https://xen-orchestra.com/docs/updater.html).
+Make sure that your appliance is [properly registered](installation#registration) and [up-to-date](https://xen-orchestra.com/docs/updater.html).
```
```suggestion
Make sure that your appliance is [properly registered](installation#registration) and [up-to-date](https://docs.xen-orchestra.com/updater.html).
```

thomas-dkmt replied:
Fixed by https://github.com/vatesfr/xen-orchestra/pull/8558/commits/90a5ddb99d90c490e321dd5bc349851f9b3f7977

thomas-dkmt committed:
doc(XO): Fix broken links to plugin documentation

Previously, the plugin documentation resided on a standalone page that wasn't listed in the sidebar and duplicated content from the Architecture page's plugin section.
This standalone Plugins page was removed in a previous commit (https://github.com/vatesfr/xen-orchestra/pull/8539), causing several links to break.
This commit updates those links to point to the correct location.

Signed-off-by: Thomas Moraine <thomas.moraine@vates.tech>
